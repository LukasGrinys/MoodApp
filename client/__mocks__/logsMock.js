export const testLogs = [
    {
        id: '001',
        rating: 10,
        date: "TEST-DATE",
        text: "Test text"
    },
    {
        id: '002',
        rating: 9,
        date: "TEST-DATE",
        text: "Test text"
    },
    {
        id: '003',
        rating: 8,
        date: "TEST-DATE",
        text: "Test text"
    },
    {
        id: '004',
        rating: 7,
        date: "TEST-DATE",
        text: "Test text"
    }
];

export const testLog = {
    id: '001',
    rating: 10,
    timing: 'Afternoon', 
    date: "TEST-DATE",
    text: "Test text"
}