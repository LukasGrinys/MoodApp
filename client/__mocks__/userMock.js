export const testUser = {
    id: '001',
    firstName: 'test',
    lastName: 'test',
    email: 'test@test.com'
}