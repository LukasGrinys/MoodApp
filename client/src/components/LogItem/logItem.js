import React from 'react';
import styles from './logItem.module.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faHourglass, faCalendar } from '@fortawesome/free-solid-svg-icons';

const LogItem = ({date, timing, text, rating}) => {
    return (
        <div className={styles.item}>
            <div className={styles.top_line}>
                <div className={styles.rating_box}>
                    {rating}
                </div>
                <div>
                    <FontAwesomeIcon icon={faHourglass} className={styles.icon}/>
                    {timing}
                </div>
                <div className={styles.date}>
                    <FontAwesomeIcon icon={faCalendar} className={styles.icon}/>
                    {date}
                </div>
            </div>
            <div className={styles.text}>
                {text}
            </div>
        </div>
   );
};

export default LogItem