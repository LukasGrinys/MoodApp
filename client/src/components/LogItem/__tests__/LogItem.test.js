import React from 'react';
import LogItem from '../logItem';
import { shallow } from 'enzyme';
import { testLog } from '../../../../__mocks__/logsMock';

const wrapper = shallow(<LogItem {...testLog}/>)

test('Renders properly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('Has rating box with log rating in it', () => {
    const ratingBox = wrapper.find('.rating_box');
    expect(ratingBox.length).toBe(1);
    expect(ratingBox.text()).toBe(testLog.rating.toString());
});

test('Has text div with log message in it', () => {
    const text = wrapper.find('.text');
    expect(text.length).toBe(1);
    expect(text.text()).toBe(testLog.text);  
});