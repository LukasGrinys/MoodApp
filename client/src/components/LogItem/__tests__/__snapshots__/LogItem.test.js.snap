// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders properly 1`] = `
<div
  className="item"
>
  <div
    className="top_line"
  >
    <div
      className="rating_box"
    >
      10
    </div>
    <div>
      <FontAwesome
        className="icon"
        name="hourglass"
      />
      Afternoon
    </div>
    <div
      className="date"
    >
      <FontAwesome
        className="icon"
        name="calendar"
      />
      TEST-DATE
    </div>
  </div>
  <div
    className="text"
  >
    Test text
  </div>
</div>
`;
